libraries/Microsoft.Bot.Builder.Dialogs.Debugging/Protocol.cs

```diff
@@ -289,6 +289,8 @@ public class Range : Reference
 
             public string More { get; set; }
 
+            public JToken Designer { get; set; }
```

Comment:
I think this is a feature for Composer which inject a designer field, right? 

I think it will do the fix, but is there an option that don't require hard-coding this field name in the protocol here? 

Because, at Composer side, i don't think we are so sure about this designer field actually. I mean,  it was helpful for some task at this stage in Composer, but we are very aware of that, it makes the json less readable, and also have impacts around lg, lu files. 

Anyhow, that's the reason we use "$designer", so, is it possible, we do a generic solution that preserve any fields starts with "$"?

Reply:
I agree with the sentiment.  Ideally, the json items (e.g. dialogs or recognizers) would have an identifier field baked into the file format, instead of the adaptive runtime generating one dynamically.  The $designer.id field is just a proxy for that identifier.